mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -154,7 +154,7 @@ func resourceSqlDatabaseInstance() *schema.Resource {
 								Schema: map[string]*schema.Schema{
 									"domain": {
 										Type:        schema.TypeString,
-										Required:    true,
+										Optional:    true,
```

Comment:
I would suggest that we keep the requirement of this field, as to prevent users from specifying an empty block in Terraform, since `domain` is the only subfield, unless `active_directory_config{}` is a config that needed to be send intentionally.

Reply:
I think we should allow active_directory_config{} as, at the moment, the API is incorrectly creating this active_directory_config block for database instances that do not need this block. If we make the domain field required, then it will be the same issue as for the bucket field below.

Reply:
what does an empty block `active_directory_config{}` mean without `domain` set up, is there some default domain will be used?

Reply:
No default value. It just gets ignored if it's empty. Backend has it as optional so just syncing it here.

Reply:
If `active_directory_config{}` is sent to API, does it mean different thing than no `active_directory_config` at all is sent?

Reply:
No difference. The frontend (Pantheon) is adding this block during instance creation. So it is causing issues in Kubernetes Config Connector, which saves the values from a GET request on such a database instance and re-uses those values for PUT request. Are there any alternatives like having some default value if we use optional?